File: dsa_js/selection_sort.js
function selectionSort(arr) {
    for (let i = 0; i < arr.length; i++) {
        let minValIndex = i;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[minValIndex]) {
                minValIndex = j;
            }
        }
        let temp = arr[i];
        arr[i] = arr[minValIndex];
        arr[minValIndex] = temp;
    }
    return arr;
}

const list = [7, 12, 9, 4, 11];
console.log('Actual array:', list);
console.log('Sorted array:', selectionSort(list));
console.log('Sorted array:', list);

File: dsa_python/selection_sort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]
    return arr


my_array = [64, 34, 25, 12, 22, 11, 90, 5]
print('Actual array is:', my_array)
print('Sorted array is:', selection_sort(my_array))
print('Sorted array is:', my_array)
